### DISCUSSION TITLE
How do we read files in serverless of Golang?

### DISCUSSION THREAD

**Discussion by darksun113** [vercel/community#873] [Help] [answered]
I saw some examples. For Python and Node.js, we can read files for relative path (https://vercel.com/docs/runtimes#advanced-usage/advanced-python-usage)

I tried to do the same thing in Go. But unfortunately I couldn't figure out which is the correct path to store the data files.

Here are the codes;

dir, err := filepath.Abs("data")
	if err != nil {
		galleryLogger.Fatal("get path error", zap.Error(err))
	}
	fileInfos, err := ioutil.ReadDir(dir)
	if err != nil {
		galleryLogger.Fatal("read dir error", zap.String("dir", dir), zap.Error(err))
	}

And the errors:

 Runtime exited with error: exit status 1
Runtime.ExitError
{"level":"fatal","ts":1663302896.9636867,"caller":"api/gallery.go:45","msg":"read dir error","dir":"/var/task/data","error":"open /var/task/data: no such file or directory"

Any help would be appreciated!

**Comment by amyegan** [comment]
Thanks for reaching out @darksun113! I have not worked with Go personally, but I'll ask around to try and find you an answer. In the meantime, maybe someone else in the community will have the answer and share it here.

**Comment by craigandrews** [comment, marked as answer]
The way to do it is to use the `includeFiles` directive in `vercel.json`. Note, however, that the files included are relative to the Go file, not the repository root. For example, this config:

```
{
  "functions": {
    "api/*.go": {
      "includeFiles": "data/*"
    }
  }
}
```

Would allow you to access the file `accessible.txt` but not the file `inaccessible.txt`:

```
api/
   handler.go
   data/
       accessible.txt
data/
   inaccessible.txt
```

**Reply by darksun113** [reply]
It worked! That's really helpful. Thanks.

**Comment by darksun113** [comment]
> Thanks for reaching out @darksun113! I have not worked with Go personally, but I'll ask around to try and find you an answer. In the meantime, maybe someone else in the community will have the answer and share it here.

Thanks! I have found the solution
